### PR TITLE
Base chain registry

### PR DESCRIPTION
**Pull request by ERussel**
- base chain registry classes implementation add
- concrete implementation for connection pool add

### CODE REVIEW

File: fearless/Common/Services/ChainRegistry/ChainSyncService.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        repository: AnyDataProviderRepository<ChainModel>
+    ) {
+        self.url = url
+        self.dataFetchFactory = dataFetchFactory
+        self.repository = repository
+    }
+}
+
+extension ChainSyncService: ChainSyncServiceProtocol {
+    func syncUp() {}
```
Comment (ef1rspb) [inline fearless/Common/Services/ChainRegistry/ChainSyncService.swift:25]: // TODO?

File: fearless/Common/Services/ChainRegistry/ConnectionPool/ConnectionPool.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+protocol ConnectionPoolProtocol {
+    func setupConnection(for chain: ChainModel) throws -> JSONRPCEngine
```
Comment (ef1rspb) [inline fearless/Common/Services/ChainRegistry/ConnectionPool/ConnectionPool.swift:4]: `-> ChainConnection`

File: fearless/Common/Services/ChainRegistry/RuntimeProviderPool/RuntimeProviderPool.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+}
+
+extension RuntimeProviderPool: RuntimeProviderPoolProtocol {
+    func getWalletRuntime(for chain: ChainModel) -> RuntimeProviderProtocol {
+        runtimeProviderFactory.createRuntimeProvider(for: chain)
+    }
+
+    func getFullRuntime(for chain: ChainModel) -> RuntimeProviderProtocol {
+        runtimeProviderFactory.createRuntimeProvider(for: chain)
```
Comment (ef1rspb) [inline fearless/Common/Services/ChainRegistry/RuntimeProviderPool/RuntimeProviderPool.swift:24]: implementation is identical to `getWalletRuntime`

File: fearless/Common/Services/ChainRegistry/RuntimeSyncService.swift [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import Foundation
+import RobinHood
+
+protocol RuntimeSyncServiceProtocol {
+    func register(chain: ChainModel, with connection: JSONRPCEngine)
```
Comment (ef1rspb) [inline fearless/Common/Services/ChainRegistry/RuntimeSyncService.swift:5]: `with connection: ChainConnection)`

File: fearless/Common/Services/ChainRegistry/ChainSyncService.swift (hunk 1 above)
Comment (ERussel) [reply fearless/Common/Services/ChainRegistry/ChainSyncService.swift]: Yep, marked all unimplemented parts with TODO providing task number

File: fearless/Common/Services/ChainRegistry/ConnectionPool/ConnectionPool.swift (hunk 2 above)
Comment (ERussel) [reply fearless/Common/Services/ChainRegistry/ConnectionPool/ConnectionPool.swift]: fixed

File: fearless/Common/Services/ChainRegistry/RuntimeProviderPool/RuntimeProviderPool.swift (hunk 3 above)
Comment (ERussel) [reply fearless/Common/Services/ChainRegistry/RuntimeProviderPool/RuntimeProviderPool.swift:26]: Yep, marked all unimplemented parts with TODO providing task number

File: fearless/Common/Services/ChainRegistry/RuntimeSyncService.swift (hunk 4 above)
Comment (ERussel) [reply fearless/Common/Services/ChainRegistry/RuntimeSyncService.swift]: fixed

**Pull request merged**